client/cypress/integration/visualizations/pivot_spec.js

```diff
@@ -73,6 +74,36 @@ describe('Pivot', () => {
     });
   });
 
+  it('updates the visualization when results change', function () {
+    const options = {
+      aggregatorName: 'Count',
+      data: [], // force it to have a data object, although it shouldn't
```

Comment:
That was the way I found to test what will happen in UI when the object has a 'data'.
It's impracticable to use the UI to configure Pivot Table (the entire thing is drag&drop).